### DISCUSSION TITLE
What's the current state of local-cache cross-tab synchronization?

### DISCUSSION THREAD

**Discussion by FezVrasta** [urql-graphql/urql#996] [Q&A]
Does urql provide any way to sync local-cache across different tabs of the same application? Is there any example available?

Thanks!

**Comment by kitten** [comment]
That’s an interesting question. Personally I’d opt for synchronizing responses as they come back from the API which could maybe be a `BroadcastChannel` on which the tab that actively makes a request publishes the result to, which then other tabs (and hence other instance of e.g. Graphcache can pick up)

They could also elect a leader that actually makes the request if any duplicates occur.

I’d imagine hence that this would be an additional exchange that can be put in front of the `fetchExchange` to reduce complexity on the caching side.

**Reply by FezVrasta** [reply]
Thanks for the quick reply! How would your idea work with local cache mutations though?

**Reply by kitten** [reply]
As those are always based on raw data, especially in Graphcache, we don't have to react to anything but results. Basically we reap the benefits of encouraging a cache that reflects server state at all times.
